spec/operators/toPromise-spec.ts

```diff
@@ -1,42 +1,79 @@
+/** @prettier */
 import { expect } from 'chai';
-import { of, EMPTY, throwError, config } from 'rxjs';
+import { config } from 'rxjs';
+import { TestScheduler } from 'rxjs/testing';
+import { observableMatcher } from '../helpers/observableMatcher';
 
 /** @test {toPromise} */
 describe('Observable.toPromise', () => {
-  it('should convert an Observable to a promise of its last value', (done: MochaDone) => {
-    of(1, 2, 3).toPromise(Promise).then(x => {
-      expect(x).to.equal(3);
-      done();
+  let rxTestScheduler: TestScheduler;
+
+  beforeEach(() => {
+    rxTestScheduler = new TestScheduler(observableMatcher);
+  });
+
+  it('should convert an Observable to a promise of its last value', (done: Mocha.Done) => {
+    rxTestScheduler.run(({ cold, expectSubscriptions }) => {
+      const source = cold('12(3|)');
+      const subs = '       ^-!';
```

Comment:
We need to be very careful replacing non-marble sources with marble sources, as they're sometimes not the same. One problem is that with marble sources, it's not possible to represent completion _within_ subscribe - see https://github.com/ReactiveX/rxjs/issues/5523.

So, to be sure that we are not changing tests, I don't think we should replace _any_ non-marble sources with marble sources.

Reply:
Ok, I was suspecting that this is something that might lead to some issues.

Looking at the tests it seems that none of the `toPromise` tests should be converted to marble tests. Therefore also the introduction of run mode does not make sense IMO. You could just remove the last commit of the PR (that is the one that introduces the marbles) or if you would like I could do that an rephrase the PR description.

Reply:
I'll just remove the last commit, I think. Thanks.